net/protocol.h

```diff
@@ -0,0 +1,58 @@
+#ifndef __PROTOCOL_H
+#define __PROTOCOL_H
+
+/**
+ * @defgroup    net_protocol      Protocol
+ * @ingroup     net
+ * @brief       Protocol definitions
+ *
+ * @{
+ * @file
+ * @author Geovane Fedrecheski <geovane.fedrecheski@inria.fr>
+ * @copyright Inria, 2024-now
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <nrf.h>
+
+//=========================== variables ========================================
+
+typedef enum {
+    DOTLINK_PACKET_TYPE_BEACON = 1,
+    DOTLINK_PACKET_TYPE_JOIN_REQUEST = 2,
+    DOTLINK_PACKET_TYPE_JOIN_RESPONSE = 3,
+    DOTLINK_PACKET_TYPE_INFRASTRUCTURE_DATA = 8,
+    DOTLINK_PACKET_TYPE_EXPERIMENT_DATA = 9,
+} dl_packet_type_t;
+
+// general packet header
+typedef struct {
+    uint8_t version;
+    dl_packet_type_t type;
+    uint64_t dst;
+    uint64_t src;
+} dl_packet_header_t;
```

Comment:
by convention, use suffix `_ht` for all headers, `_t` for other types

Reply:
This sounds a bit too much to me.
I never saw this convention, and also could not find about it online.
Also, all other non-header custom types tend to be `*_vars_t` which clearly differentiate them.
I would prefer to keep it as is.